### ISSUE TITLE
Attempt to call non-existing CPU constructor crashes the typechecker

### ISSUE THREAD

**Issue by peterwvj** [overturetool/overture#698]
### Description

Attempting to call a CPU constructor that only takes a single argument crashes the typechecker.

### Steps to Reproduce

1. Create a file S.vdmrt with the following content

```
system S
instance variables
public static c : CPU := new CPU(22e6);
end S
```
2. Type-check the model, for example by executing

```bash
java -jar Overture.jar -vdmrt S.vdmrt
```

**Expected behavior:**

Expected an error similar to the following

```
Error 3134: Class has no constructor with these parameter types in 'S' (S.vdmrt) at line 5:26
Called: CPU(nat1)
```

**Actual behavior:**

The typechecker crashes with an `IndexOutOfBoundsException`:

```
Parsed 1 class in 0.034 secs. No syntax errors
java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
Type checked 3 classes in 0.017 secs. Found 1 type error
```

**Reproduces how often:**

Every time.

### Versions

Overture 2.6.4, Ubuntu 18.04.

### Additional Information

VDMJ does *not* have this problem.

**Comment by nickbattle** [comment]
This is something that was added to Overture (RT) to do with clock frequencies. It looks like there is a simple error in argument indexing, which would go unnoticed if you provided two arguments as expected. I've pushed a fix to ncb/development. So now the example says:
```
Error 3305: CPU frequency to slow: 0.0 Hz in 'S' (test.vpp) at line 3:15
Error 3134: Class has no constructor with these parameter types in 'S' (test.vpp) at line 3:26
Called: CPU(nat1)
```
That "too slow" check is triggered because there is no frequency argument supplied. But it's unusual that this is being done before the type check of the parameters. At least it doesn't crash now :-)
